contrib/IECoreUSD/test/IECoreUSD/USDSceneTest.py

```diff
@@ -474,6 +474,60 @@ def testInstancesShareHashes( self ) :
 				self.assertEqual( instance0.hash( hashType, 0 ), instance1.hash( hashType, 0 ) )
 				self.assertEqual( notInstance.hash( hashType, 0 ), instance0.hash( hashType, 0 ) )
 
+	def testInstancePrototypeHashesNotReused( self ) :
+
+		# The original intent of this test was to check that we assign consistent hashes to prototypes when
+		# opening and closing the same file ( which triggers USD to randomly shuffle the prototype names ).
+
+		# The solution we've ended up with is instead of assigning consistent hashes, each instance of the
+		# file gets it's own unique hashes, and is basically treated separately. This allows us to just
+		# use the prototype names in the hash, since we force the hash to be unique anyway.
+		
+		usedHashes = set()
+
+		for i in range( 100 ):
+			scene = IECoreScene.SceneInterface.create(
+				os.path.dirname( __file__ ) + "/data/severalInstances.usda",
+				IECore.IndexedIO.OpenMode.Read
+			)
+
+			instance0 = scene.child( "instance0" )
+			instance0Child = instance0.child( "world" )
+			instance10 = scene.child( "instance10" )
+			instance10Child = instance10.child( "model" ).child( "assembly" )
+
+			h1 = instance0Child.hash( scene.HashType.TransformHash, 1 )
+			h2 = instance10Child.hash( scene.HashType.TransformHash, 1 )
+
+			self.assertNotEqual( h1, h2 )
+
+			self.assertFalse( h1 in usedHashes )
```

Comment:
It's a small thing, but `assertNotIn()` gives more informative error messages than `assertFalse()`.